app/views/bots/index.html.erb

```diff
@@ -17,8 +17,10 @@
             <% if bot.repository.present? %><p class="text-muted">Repository: <%= repo_link(bot) %></p><% end %>
             <p><strong><%= link_to "Instances", url_for(controller: :bot_instances, action: :index, bot_id: bot.id) %></strong></p>
             <ul class="bot-instance-ul<%= " no-collaborators" unless bot.collaborators.any? %>">
-              <% bot.bot_instances.order(:priority).where("last_ping > ?", 1.hour.ago).each do |instance| %>
+              <% if bot.bot_instances.order(:priority).where("last_ping > ?", 1.hour.ago).each do |instance| %>
                 <%= render 'bot_instances/list_item', :instance => instance %>
+              <% end.length == 0 %>
```

Comment:
Clever, but a little hard to read. I'd go with a quick `if bot.bot_instances.where("last_ping > ?", 1.hour.ago).any?` check instead.